src/utils/LinearAlg2D.cpp

```diff
@@ -9,6 +9,8 @@
 
 #include "IntPoint.h" // dot
 
+#define SQRT_LLONG_MAX_FLOOR 3037000499
```

Comment:
Use constexpr.

Unfortunately sqrt is not a constexpr enabled function, and will not even be in C++20, so we can't do `constexpr size_t sqrt_long_max_floor = std::sqrt(std::numeric_limits<coord_t>::max());`, but we _can_ do `constexpr size_t sqrt_long_max_floor = 3037000499;`.